ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -625,9 +652,9 @@ private void performHardDeletes(List<MessageInfo> messageInfoList) throws StoreE
           throw new StoreException("Aborting hard deletes as store is shutting down",
               StoreErrorCodes.Store_Shutting_Down);
         }
+        diskIOScheduler.getSlice(HARD_DELETE_CLEANUP_JOB_NAME, HARD_DELETE_CLEANUP_JOB_NAME, logWriteInfo.size);
```

Comment:
any reason to change the ordering here? Throttle the hard delete I/O ?

Reply:
the reason for changing the order is to use diskIOScheduler.getSlice to throw an exception before  modify the log file. The reason why I want to throw an exception is to mock the situation when we persist the HardDeleteRecoveryRange but fail to actually hard delete the records in the log file. Under such situation, we can test if the recovery actually works or not.